react/features/remote-video-menu/components/native/MuteButton.js

```diff
@@ -1,14 +1,50 @@
 // @flow
 
 import { translate } from '../../../base/i18n';
+import { isLocalParticipantModerator } from '../../../base/participants';
 import { connect } from '../../../base/redux';
 
-import AbstractMuteButton, { _mapStateToProps } from '../AbstractMuteButton';
+import AbstractMuteButton, {
+    _mapStateToProps as _abstractMapStateToProps,
+    type Props as AbstractProps
+} from '../AbstractMuteButton';
+
+type Props = AbstractProps & {
+
+    /**
+     * True if the local participant is a moderator.
+     */
+    _isModerator: boolean,
+}
+
+/**
+ * Implements a button to mute the remote participant.
+ */
+class MuteButton extends AbstractMuteButton<Props> {
+    /**
+     * Overrides {@code AbstractButton#_shouldRender}.
+     *
+     * @inheritdoc
+     */
+    _shouldRender() {
+        return this.props._isModerator;
+    }
+}
 
 /**
- * We don't need any further implementation for this on mobile, but we keep it
- * here for clarity and consistency with web. Once web uses the
- * {@code AbstractButton} base class, we can remove all these and just use
- * the {@code AbstractMuteButton} as {@MuteButton}.
+ * Maps part of the Redux state to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Object} ownProps - Properties of component.
+ * @returns {{
+ *     _tileViewEnabled: boolean
```

Comment:
I still don't like these types of @returns docs because they get outdated so easily.

Reply:
Yeah, right ? The props are described above by the `Props` type anyway. Actually it could be just `@returns {Props}`

Reply:
I fully agree! I change it now to {Props}, that is partially true, but I consider it better than just using {Object}

Reply:
out of curiosity: why is it partially true ? is it that not all props may have been mapped ?

Reply:
As far as I know, not. It's the ```mergeProps``` that does the merging, but we rarely use that. So the ```mapStateToProps``` basically returns a partial ```Props``` object